### PR TITLE
Add error message if webgpu is not supported

### PR DESCRIPTION
**Pull request by Vrixyz**
image

We could also display a message if javascript is not enabled, let me know if that's interesting.

### CODE REVIEW

File: static/demos/wgsparkl2d/index.html [hunk 1]
```diff
@@ -3,13 +3,41 @@
...
+            messageContainer.style.color = "red";
+            messageContainer.style.fontWeight = "bold";
+            messageContainer.style.margin = "20px";
+            messageContainer.innerHTML = message;
+            document.body.appendChild(messageContainer);
+        }
+
+        async function check_webgpu_compatibility() {
+            if (!navigator.gpu) {
+                displayErrorHTML("WebGPU not supported. <a href='https://caniuse.com/webgpu'>More info</a>")
```
Comment (Vrixyz) [inline static/demos/wgsparkl2d/index.html:19]: Linking to caniuse can be controversial, alternatives would be:
- https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API#browser_compatibility is more informative but more developer-centric and "biased" towards mozilla.
- No link, leave it as an exercise for the reader

Comment (Vrixyz) [reply static/demos/wgsparkl2d/index.html:19]: I'll also take suggestions on wording: "This browser does not support WebGPU." might be better.

**Review by sebcrozet — CHANGES_REQUESTED** [review]
Sweet, thank you!

Suggestion for the text:

```html
WebGPU is not supported on this browser.  <-- Put that line in bold red
See <a href='https://caniuse.com/webgpu'>caniuse.com</a> for a list of compatible browsers.
```

In addition, can you make it centered (vertically and horizontally), with a logo of wgmath (like https://github.com/dimforge/wgmath.rs/blob/main/static/img/wgmath_logo.svg) above the error message?

File: static/demos/wgsparkl2d/index.html (hunk 1 above)
Comment (sebcrozet) [reply static/demos/wgsparkl2d/index.html:17]: A link to caniuse sounds fine to me.

**Comment by Vrixyz** [comment]
Sure! I pushed those changes:

image

**Review by sebcrozet — APPROVED** [review]
Fantastic, thanks!

**Pull request merged**